cirq/circuits/moment.py

```diff
@@ -111,3 +111,7 @@ def __repr__(self):
 
     def __str__(self):
         return ' and '.join(str(op) for op in self.operations)
+
+    def with_qubits_mapped(self, qubit_map: Dict[ops.QubitId, ops.QubitId]):
```

Comment:
Call this `transform_qubits` and take a lambda instead of a dictionary, to be consistent with `Operation.transform_qubits`.

Then use `op.transform_qubits(func)` in your loop, instead of `with_qubits`.

The shorthand for passing a dictionary as a transformation is `func=my_dict.__getitem__`.